xbmc/windowing/osx/WinSystemOSX.mm

```diff
@@ -919,14 +920,15 @@ static void DisplayReconfigured(CGDirectDisplayID display,
     UnblankDisplays();
   }
 
-  //DisplayFadeFromBlack(fade_token, needtoshowme);
-
-  m_fullscreenWillToggle = true;
   // toggle cocoa fullscreen mode
-  if ([m_appWindow respondsToSelector:@selector(toggleFullScreen:)])
+  if (fullScreenState != m_bFullScreen &&
+      [m_appWindow respondsToSelector:@selector(toggleFullScreen:)])
```

Comment:
`respondsToSelector` check looks like some legacy leftover, this function is available since 10.7 - you can call it directly in the body.

Reply:
done

Reply:
please also replace

```objc
[m_appWindow performSelectorOnMainThread:@selector(toggleFullScreen:)
                                   withObject:nil
                                waitUntilDone:YES];
```

with `[m_appWindow toggleFullScreen:nil]` - that's what I meant by

> you can call it directly in the body

Reply:
@kambala-decapitator :
<img width="532" alt="image" src="https://user-images.githubusercontent.com/7375276/208234954-166bddc9-9845-4e45-8dfa-37439f3bede7.png">

Reply:
aha! I was inattentive. No change needed then. Thanks!